eatngo-core/src/main/kotlin/com/eatngo/store/serviceImpl/StoreServiceImpl.kt

```
@@ -19,59 +18,71 @@ import java.time.LocalDateTime
-    override fun updateStoreOnlyStatus(id: Long, newStatus: String): StoreDto {
-        val existingStore = storePersistence.findById(id)
-            ?: throw StoreException.StoreNotFound(id)
-        val enumStatus = try {
-            StoreEnum.StoreStatus.valueOf(newStatus.uppercase())
+    override fun updateStoreStatus(id: Long, newStatus: String, storeOwnerId: Long): Store {
+        val existingStore = storePersistence.findById(id).orThrow { StoreException.StoreNotFound(id) }
+        existingStore.requireOwner(storeOwnerId)
+
+        val status = try {
+            StoreEnum.StoreStatus.valueOf(newStatus.trim().uppercase())
         } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)